src/main/java/frc/robot/auton/ParallelActionRunner.java

```diff
@@ -0,0 +1,24 @@
+package frc.robot.auton;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ParallelActionRunner {
+
+    List<AutonAction> actions = new ArrayList<>();
+
+    public void addActionToRun(AutonAction action) {
+        actions.add(action);
+        action.initiate();
+    }
+
+    public void onEveryFrame() {
+        for (int i = 0; i < actions.size(); i++) {
+            AutonAction action = actions.get(i);
+            if (action.isDone()) {
+                actions.remove(i);
+                i--;
+            }
+        }
+    }
```

Comment:
Instead of doing a loop with indexes here, you could do something like 
```java
for (AutonAction action : actions) {
  if (action.isDone()) {
    actions.remove(action);
  }
}
```

This way you don't have to deal with indexes at all

Reply:
So turns out I'm wrong and Java is weird. You can do it in a way with iterators like this

https://stackoverflow.com/questions/1196586/calling-remove-in-foreach-loop-in-java

But if you guys aren't comfortable with that then you can just keep it like you have it!